### PR TITLE
#1025: Implement `Overflow` (closes #1025)

### LINKED ISSUE #1025: Implement `ComponentOverflow`
## requirements
as we've done with `TextOverflow`, it would be useful to have a `ComponentOverflow` that checks if the component is not entirely displayed and, in this case, shows a custom placeholder on the right and renders the entire component in a `Popover` when hovered.

## specs
- move logic from `TextOverflow` to generic `Overflow` component
- use it to implement `ComponentOverflow`

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Closes #1025

### tests performed
tested a POC on AlinityPRO -> worked fine

tested with following example:
[https://gyazo.com/159b1f888d8fbca32144f415761ae9e2](https://gyazo.com/159b1f888d8fbca32144f415761ae9e2)

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

**Comment by FrancescoCioria** [comment]
@giogonzo please don't merge it yet

File: src/Overflow/Overflow.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+export const Props = {
+  children: t.ReactChild,
+  onChange: t.Function,
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+/**
+ * Util component which accepts calls a callback whenever the content starts or stop overflowing.
```
Comment (giogonzo) [inline src/Overflow/Overflow.js]: typo `accepts calls`

File: src/Overflow/Overflow.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * @param children - a function that will be called with the argument
+ * @param onChange - tooltip delay if the component is lazy
```
Comment (giogonzo) [inline src/Overflow/Overflow.js]: outdated descriptions :D

File: src/Overflow/Overflow.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+
+  onResize = () => this.verifyOverflow()
+
+  render() {
+    const { children, style, className, id } = this.props;
+
+    return (
+      <ResizeSensor debounce={10} onResize={this.onResize}>
```
Comment (giogonzo) [inline src/Overflow/Overflow.js]: magic number! Is it the same we are using in other components?

File: src/Overflow/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Overflow
+
+Util component which accepts calls a callback whenever the content starts or stop overflowing.
+
+## Props
```
Comment (giogonzo) [inline src/Overflow/README.md]: looks out of sync compared to the real props

File: src/Overflow/Overflow.js (hunk 1 above)
Comment (ecamellini) [inline src/Overflow/Overflow.js]: starts or stop**s** (s missing)

File: src/Overflow/Overflow.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        <div {...{ className, id }} style={{ ...style, width: '100%' }} ref={ref => { this.ref = ref; }}>
```
Comment (ecamellini) [inline src/Overflow/Overflow.js]: Why `div` and not `FlexView`?

Comment (ecamellini) [inline src/Overflow/Overflow.js]: The wrapper (`div`, `FlexView` etc) could be passed as a prop maybe, like it happens in other BRCs

File: src/Overflow/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+|Name|Type|Default|Description|
+|----|----|-------|-----------|
+| **children** | <code>ReactChildren</code> |  | **required**. A function that will be called with the argument |
+| **onChange** | <code>Function</code> |  | **required**. Tooltip delay if the component is lazy |
+| **verifyOverflowOn** | <code>enum("resize" &#124; "hover")</code> | <code>"resize"</code> | *optional*. Check if the content is overflowing on one of the following events: "resize" &#124; "hover" |
```
Comment (ecamellini) [inline src/Overflow/README.md]: Not a prop

File: src/Overflow/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+| **debounceVerifyOverflow** | <code>Number</code> |  | *optional*. Use this to debounce the calls to verify if the content is overflowing |
```
Comment (ecamellini) [inline src/Overflow/README.md]: Not a prop

File: src/Overflow/README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Overflow
+
+Util component which accepts calls a callback whenever the content starts or stop overflowing.
+
+## Props
+|Name|Type|Default|Description|
+|----|----|-------|-----------|
+| **children** | <code>ReactChildren</code> |  | **required**. A function that will be called with the argument |
+| **onChange** | <code>Function</code> |  | **required**. Tooltip delay if the component is lazy |
```
Comment (ecamellini) [inline src/Overflow/README.md]: "Tooltip delay if the component is lazy" and "A function that will be called with the argument" are not the right descriptions for these props

**Pull request merged**